### DISCUSSION TITLE
windows下使用pycharm进行训练出现“RANK”

### DISCUSSION THREAD

**Discussion by wenhaixi** [STVIR/pysot#513] [Q&A]
我想问下在windows下运行pysot，前面的demo，test我都跑通了，但是在运行train时候，代码是Ubuntu下的，好像是分布式的多cpu训练，运行后报错KeyError: 'RANK'，查询了discussion区 #215  #42 也并没有发现解决方案，请问如何解决的呢？

**Comment by ZhiyuanChen** [comment]
> 我想问下在windows下运行pysot，前面的demo，test我都跑通了，但是在运行train时候，代码是Ubuntu下的，好像是分布式的多cpu训练，运行后报错KeyError: 'RANK'，查询了discussion区 #215 #42 也并没有发现解决方案，请问如何解决的呢？

PySOT使用了分布式训练，这在Widnwos中暂不支持。你必须删除所有相关代码才能正常运行

**Reply by wenhaixi** [reply]
#rank, world_size = dist_init()
    #logger.info("init done") 
这边我发现main下面前两个语句是多卡的，我注释掉后可以debug了，但是为了以防万一，请问这里面还有哪里是需要注释掉的吗？

**Reply by ZhiyuanChen** [reply]
> ```
> #rank, world_size = dist_init()
> #logger.info("init done") 
> ```
> 
> 
> 这边我发现main下面前两个语句是多卡的，我注释掉后可以debug了，但是为了以防万一，请问这里面还有哪里是需要注释掉的吗？

dataloader可能也需要改动

**Reply by DDuranin** [reply]
> ```
> #rank, world_size = dist_init()
> #logger.info("init done") 
> ```
> 
> 这篇我发现主要下面的两个语句是多卡的，我注释掉后可以调试了，但是为了等万一，请问这里面有什么需要注释掉的吗？

您好 ，请问下您实现单卡训练了吗？如果方便可以留下邮箱联系吗？感谢
